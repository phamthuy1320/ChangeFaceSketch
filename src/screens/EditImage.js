import React,{useState} from 'react';
import {
    View, 
    Text, 
    Image, 
    StyleSheet,
    Dimensions,
} from 'react-native';
// import {useRoute, useNavigation} from '@react-navigation/native';
import {Header} from './commons';
import AntDesign from 'react-native-vector-icons/AntDesign';
// import SimpleLineIcons from 'react-native-vector-icons/SimpleLineIcons';
// import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import ImagePicker from 'react-native-image-crop-picker';
import {BottomPhoto} from './components/BottomBar';
import {BGCOLOR,HEADER} from './commons/Colors';
const w = Dimensions.get('window');
const photo = 'https://i.pinimg.com/564x/bd/0f/4f/bd0f4f9d99dee4fd5c3ea53facc7492f.jpg'
const EditImage = ({navigation, route}) =>{
    // const route = useRoute();
    // const navigation = useNavigation();
    const [cropedURI, setCropedURI] = useState();
    const _openCropper = (uri) =>{
        ImagePicker.openCropper({
            path: uri
          }).then(image => {
            console.log(image.path);
            setCropedURI(image.path)
          });
    }
    
    return(
        <>
            <Header 
                title = {'PHOTO'}
                left = {<AntDesign color = {HEADER.text} name = 'arrowleft' size = {22}
                    onPress = {()=>navigation.goBack()}
                />}
                right = {<View/>}
            />
            
            <View style = {styles.container}>
                <Image 
                    // source = {route?.params.source}
                    source = {{uri:cropedURI?cropedURI:route?.params?.uri||photo}}
                    style = {styles.image}
                    resizeMode = 'contain'
                />
            </View>

            <BottomPhoto
                navigation = {navigation}
                fromCamera = {route?.params?.fromCamera}
                onPressCrop = {()=>_openCropper(route?.params.uri)}
                onPressSave = {()=>navigation.navigate('Save',{uri:cropedURI?cropedURI:route?.params?.uri})}
            />
        </>
    )
}

export default EditImage;

const styles = StyleSheet.create({
    container:{
        backgroundColor:BGCOLOR,
        flex:1,
        padding:5
    },
    image:{
        width: '100%',
        height:'100%',
    }
})