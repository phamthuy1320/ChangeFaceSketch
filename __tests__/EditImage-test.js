import React from 'react';
import renderer from 'react-test-renderer';
import EditImage from '../src/screens/EditImage';

test('Edit Image Screen', () => {
  const tree = renderer.create(<EditImage/>).toJSON();
  expect(tree).toMatchSnapshot();
});