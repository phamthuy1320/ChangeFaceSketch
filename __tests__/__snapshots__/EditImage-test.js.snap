// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit Image Screen 1`] = `
Array [
  <View
    style={
      Array [
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "flexDirection": "row",
          "height": 50,
          "justifyContent": "space-between",
          "paddingHorizontal": 5,
        },
        Object {
          "backgroundColor": "#9f4347",
        },
      ]
    }
  >
    <Text
      allowFontScaling={false}
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#fff",
            "fontSize": 22,
          },
          undefined,
          Object {
            "fontFamily": "anticon",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
    <Text
      style={
        Array [
          Object {
            "textAlign": "center",
          },
          Object {
            "color": "#fff",
          },
        ]
      }
    >
      PHOTO
    </Text>
    <View />
  </View>,
  <View
    style={
      Object {
        "backgroundColor": "#f1c3aa",
        "flex": 1,
        "padding": 5,
      }
    }
  >
    <Image
      resizeMode="contain"
      source={
        Object {
          "uri": "https://i.pinimg.com/564x/bd/0f/4f/bd0f4f9d99dee4fd5c3ea53facc7492f.jpg",
        }
      }
      style={
        Object {
          "height": "100%",
          "width": "100%",
        }
      }
    />
  </View>,
  <View
    style={
      Object {
        "backgroundColor": "#9f4347",
        "flexDirection": "row",
        "justifyContent": "space-around",
        "padding": 10,
      }
    }
  >
    <View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignContent": "center",
            "alignItems": "center",
            "aspectRatio": 1,
            "backgroundColor": "#fff",
            "borderRadius": 50,
            "height": 50,
            "justifyContent": "center",
            "opacity": 1,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#9f4347",
                "fontSize": 22,
              },
              undefined,
              Object {
                "fontFamily": "anticon",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
      <Text
        style={
          Array [
            Object {
              "textAlign": "center",
            },
            Object {
              "color": "#fff",
            },
          ]
        }
      >
        Camera
      </Text>
    </View>
    <View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignContent": "center",
            "alignItems": "center",
            "aspectRatio": 1,
            "backgroundColor": "#fff",
            "borderRadius": 50,
            "height": 50,
            "justifyContent": "center",
            "opacity": 1,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#9f4347",
                "fontSize": 22,
              },
              undefined,
              Object {
                "fontFamily": "simple-line-icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
      <Text
        style={
          Array [
            Object {
              "textAlign": "center",
            },
            Object {
              "color": "#fff",
            },
          ]
        }
      >
        Crop
      </Text>
    </View>
    <View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignContent": "center",
            "alignItems": "center",
            "aspectRatio": 1,
            "backgroundColor": "#fff",
            "borderRadius": 50,
            "height": 50,
            "justifyContent": "center",
            "opacity": 1,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#9f4347",
                "fontSize": 22,
              },
              undefined,
              Object {
                "fontFamily": "Entypo",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
      <Text
        style={
          Array [
            Object {
              "textAlign": "center",
            },
            Object {
              "color": "#fff",
            },
          ]
        }
      >
        Save
      </Text>
    </View>
  </View>,
]
`;
